src/python/pants/core/util_rules/adhoc_binaries.py

```diff
@@ -104,10 +104,11 @@ async def download_python_binary(
             cp -r python "{installation_root}"
             touch "{installation_root}/DONE"
         fi
+        echo $(realpath "{installation_root}")/bin/python3
```

Comment:
Drive-by comment: Maybe double quote the expansion in case there are spaces in `installation_root`?

Reply:
```suggestion
        echo "$(realpath "{installation_root}")/bin/python3"
```